java/manager-build.xml

```diff
@@ -298,135 +302,156 @@
-    <fail>kubectl is not in the PATH. Please install kubectl first.</fail>
+  <target name="warn-if-podman-not-installed" depends="is-podman-installed" unless="podman.installed">
+    <fail>podman is not in the PATH. Please install podman first.</fail>
   </target>
 
-  <target name="deploy-kube" depends="webapp,warn-if-kubectl-not-installed" description="Deploy a new copy of SUSE Manager in a running kubernetes pod">
-    <echo message="Copying files to pod..." />
-
-    <exec failonerror="true" executable="sh">
-        <arg line="-c 'tar c -C ${build.dir}/webapp -f - --exclude=log4j2.xml . | kubectl exec -i ${kubectl.args} -- tar xf - -C ${deploy.dir}'"/>
+  <target name="check-podman-connection" depends="warn-if-podman-not-installed" description="Checks if podman system connection is set">
+    <exec executable="sh" outputproperty="podman.output">
+      <arg line="-c 'podman system connection ls -q | grep -w ${podman.connection.name}'"/>
     </exec>
 
-    <echo message="Linking the branding jar..." />
-    <exec failonerror="true" executable="kubectl">
-        <arg line="exec ${kubectl.args} -- mv ${deploy.dir}/WEB-INF/lib/java-branding.jar /usr/share/rhn/lib"/>
-    </exec>
+    <condition property="podman.connection.exists">
+      <length string="${podman.output}" when="greater" length="0"/>
+    </condition>
 
-    <exec failonerror="true" executable="kubectl">
-      <arg line="exec ${kubectl.args} -- ln -sf /usr/share/rhn/lib/java-branding.jar ${deploy.dir}/WEB-INF/lib/java-branding.jar"/>
-    </exec>
+    <echo message="Podman connection to '${podman.connection.name}' detected: ${podman.connection.exists}"/>
+  </target>
 
-    <echo message="Linking jars for Taskomatic..."/>
-    <exec failonerror="true" executable="kubectl">
-      <arg line="exec ${kubectl.args} -- ln -sf ${deploy.dir}/WEB-INF/lib/*.jar /usr/share/spacewalk/taskomatic"/>
+  <target name="add-podman-connection" depends="check-ssh-key,check-podman-connection" unless="podman.connection.exists" description="Adds podman system connection">
+    <echo message="Adding podman system connection..."/>
+    <exec executable="podman">
+      <arg line="system connection add --identity ${ssh.key} ${podman.connection.name} ssh://${deploy.user}@${podman.connection.name}.${deploy.host}"/>
     </exec>
-    <exec failonerror="true" executable="kubectl">
-      <arg line="exec ${kubectl.args} -- mv ${deploy.dir}/WEB-INF/lib/rhn.jar /usr/share/rhn/lib"/>
-    </exec>
-    <exec failonerror="true" executable="kubectl">
-      <arg line="exec ${kubectl.args} -- ln -sf /usr/share/rhn/lib/rhn.jar ${deploy.dir}/WEB-INF/lib"/>
+  </target>
+
+  <target name="remove-podman-connection" depends="warn-if-podman-not-installed" description="Removes podman system connection">
+    <echo message="Removing podman system connection..."/>
+    <exec executable="podman">
+      <arg line="system connection rm ${podman.connection.name}"/>
```

Comment:
I probably mislead you here, but the more I think about it, the more I'm convinced we don't want any `podman` or `kubectl` specific code here. We can just document how to setup the connections. Setting up a remote connection on `podman` doesn't prevent from using the local `podman` instance if the user doesn't set the `CONTAINER_CONNECTION` variable or the `--connection` parameter.